lib/push.js

```diff
@@ -0,0 +1,186 @@
+        // REMOVE APPLICATION SPECIFIC CODE FROM LIBRARY
+        // no alloy models code should be here!!
+        updateParams = {
+            notificationReceive : receivePush,
+            objectId: Ti.App.Properties.getString('com.sensimity.ti.parse_objectId'),
+            body : {
+                "channels": channels.get(),
+                "timeZone": timezone(),
+                "deviceType" : "android"
+            }
+        };
+    }
+
+    client.updatePush(updateParams, function(_response) {
+        channels.reset();
+        logger.info("client.updatePush -  " + JSON.stringify(_response));
+    }, function(_error) {
+        channels.reset();
+        logger.error("client.updatePush ERROR-  " + JSON.stringify(_error));
+    });
+}
+
+// Process incoming push notifications for ios
+function receivePush(e) {
+    logger.info('Received push: ' + JSON.stringify(e));
+
+    dispatcher.trigger("parse.push.received", e);
+}
+
+// Enable push notifications for this device
+// Save the device token for subsequent API calls
+function deviceTokenSuccess(e) {
+
+    var registerParams = {};
+
+    if (constant.OS_IOS) {
+        logger.info("retrieved token!: " + e.deviceToken);
+
+        //*******************************************************
+        // REMOVE APPLICATION SPECIFIC CODE FROM LIBRARY
+        // no alloy models code should be here!!
+        registerParams = {
+            body : {
+                "channels": channels.get(),
+                "deviceType" : "ios",
+                "deviceToken" : e.deviceToken,
+                "appIdentifier" : Ti.App.id,
+                "appName" : Ti.App.name,
+                "appVersion" : Ti.App.version,
+                "timeZone": timezone(),
+                "installationId" : Ti.Platform.createUUID()
+            }
+        };
+    } else {
+        logger.debug("registering the android device for push");
+
+        //*******************************************************
+        // REMOVE APPLICATION SPECIFIC CODE FROM LIBRARY
+        // no alloy models code should be here!!
+        registerParams = {
+            notificationReceive : receivePush,
+            body : {
+                "channels": channels.get(),
+                "timeZone": timezone(),
+                "deviceType" : "android"
+            }
+        };
+    }
+
+    client.registerPush(registerParams, function(_response) {
+        channels.reset();
+        logger.info("client.registerPush -  " + JSON.stringify(_response));
+
+        Ti.App.Properties.setString('com.sensimity.ti.parse_objectId', _response.objectId);
+
+    }, function(_error) {
+        channels.reset();
+        logger.error("client.registerPush ERROR-  " + JSON.stringify(_error));
+    });
+
+}
+
+function deviceTokenError(e) {
+    logger.info('Failed to register for push notifications! ' + e.error);
+}
+
+module.exports = {
+    registerPush : registerPush,
+    updatePush : updatePush
+};
```

Comment:
Is there any reason to not end with a new-line?
I found a article about ending with new-lines in javascripts: http://evanhahn.com/newline-necessary-at-the-end-of-javascript-files/. What's your opinion?